### ISSUE TITLE
How to send conditional http status code from OPA to the caller 

### ISSUE THREAD

**Issue by kumar-subashp** [open-policy-agent/opa#2821]
This is a query not an issue.

I am implementing RBAC using OPA, like if a user has so and so roles should allow by returning http status code as 200(OK) else deny that's 403(Forbidden)

Code snippet :
```
package authz

import input
import data.roles

default allow = {
  "allowed": false,
  "body": "Forbidden Request",
  "http_status": 403
}

allow = response {
   contains(roles, input.role)
   response := { "allowed": true }
}
```
in this case value returning part of response body, but I want conditional Http Status code part of response object not in body. please suggest.

**Comment by patrick-east** [comment]
>  I want conditional Http Status code part of response object not in body

Can you elaborate on that more? I'm not sure I understand what you mean by being part of the response object and not in the body. ie do you mean to have the OPA HTTP API return a custom http status? Or the `response` object in the rego code?

**Comment by kumar-subashp** [comment]
Hi Patrick,

My bad, let me rephrase.. mainly HTTP Response Message will have mainly following sections

   **Status Line** eg: HTTP/1.1 200 OK
   **Headers**
   **Message Body**

I was mainly looking for Status Line, is it possible to set the http status code conditionally in rego.

**Comment by patrick-east** [comment]
The HTTP status codes returned by OPAs APIs are reflecting the status of the evaluation itself (eg, was there an internal error or something while evaluating the policy or was the evaluation successful) versus what logical result comes out of the policy. There isn't a way to affect the status or headers from the policy result.

**Comment by kumar-subashp** [comment]
thanks for the clarification
